### PR TITLE
Test/add cg table test

### PR DESCRIPTION
**Pull request by ramenhog**
## Issue Number
#69 

## Purpose/Implementation Notes
Add unit tests for `cgTable`:
* displays correct number of columns based on column config
* displays correct number of rows based on data passed into component
* can detect if all results are selected
* properly selects all results
* properly unselects all results
* sets sort variables correctly when sorting is called

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests
* tests pass locally

## Checklist

_Put an `x` in the boxes that apply._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/unit/components/common/cgTable/cgTable_spec.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    expect(areResultsAllSelected).toBe(true);
+  });
+
+  it("unselects all results", () => {
+    controller.selectAllResults(false);
+    const areResultsAllSelected = parentScope.data.every(
+      result => result.isSelected === true
+    );
+    expect(areResultsAllSelected).toBe(false);
+  });
```
Comment (mhuyck) [inline test/unit/components/common/cgTable/cgTable_spec.js]: There is a small logic error here, meaning that this isn't quite testing what we're trying to test.

Let's say we have five results, labeled `A`, `B`, `C`, `D`, and `E`.

The test for each of the `result.isSelected` lines (129) will be `false` (if all goes well in `selectAllResults(false)`), so the summary at `parentScope.data.every()` will give us, effectively:

> `areResultsAllSelected` = `false` && `false` && `false` && `false` && `false`

This is, indeed, `false`. But what if `selectAllResults(false)` is broken, and it stops doing its job after the first entry? Then we have the following as a possible result:

> `areResultsAllSelected` = `false` && `false` && `false` && `false` && `true`

This is still `false`.

What we really want is a test that mirrors the "selects all results" test in a different way. I think the test should be something like:

```javascript
    const areResultsAllNotSelected = parentScope.data.every(
      result => result.isSelected === false
    );
    expect(areResultsAllNotSelected).toBe(true);
```

File: test/unit/components/common/cgTable/cgTable_spec.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  it("sets sortType and sortReverse correctly when sorting", () => {
+    controller.sortColumn("Name");
+    expect(controller.sortType).toEqual("Name");
+    expect(controller.sortReverse).toBe(true);
+  });
```
Comment (mhuyck) [inline test/unit/components/common/cgTable/cgTable_spec.js]: Hmm. Is it really desired behavior that the sort order reverses *every time* the sort column is changed? I could see this happening if you click *again* on the currently-sorted column (and that is pretty standard), but that is not the behavior coming from this test nor from the implementation in `cgTable.component.js`, line 46.

If flipping the sort order each time is really what should happen, then this test (and the component) works well. If not, this is a good opportunity to fix a sorting bug.

**Comment by ramenhog** [comment]
@mhuyck nice catches! Just fixed them and this is ready for re-review, thanks! 👍

File: test/unit/components/common/cgTable/cgTable_spec.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  it("displays the correct number of columns", () => {
+    let columnCells = angular.element(element[0].querySelectorAll("th"));
+    expect(columnCells.length).toEqual(parentScope.columns.length + 1);
+  });
+
+  it("displays the correct number of rows", () => {
+    let tableRows = angular.element(element[0].querySelectorAll("tbody tr"));
+    expect(tableRows.length).toEqual(parentScope.data.length);
+  });
```
Comment (mhuyck) [inline test/unit/components/common/cgTable/cgTable_spec.js]: 👍

File: test/unit/components/common/cgTable/cgTable_spec.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    expect(areResultsAllSelected).toBe(true);
+  });
+
+  it("unselects all results", () => {
+    controller.selectAllResults(false);
+    const areResultsAllUnselected = parentScope.data.every(
+      result => result.isSelected === false
+    );
+    expect(areResultsAllUnselected).toBe(true);
+  });
```
Comment (mhuyck) [inline test/unit/components/common/cgTable/cgTable_spec.js]: 👍

**Pull request merged**